src/config.rs

```diff
@@ -218,16 +219,45 @@ impl WriteOptions {
     }
 }
 
+/// Configuration for client put operations. `PutOptions` is supplied for each
+/// row inserted. This differs from [`WriteOptions`] in that a write may encompass
+/// multiple puts (such as the case with batched writes)
+#[derive(Clone, Default)]
+pub struct PutOptions {
+    /// The time-to-live (ttl) for this insertion. If this insert overwrites an existing
+    /// database entry, the TTL for the most recent entry will be canonical.
+    ///
+    /// Default: the TTL configured in DbOptions when opening a SlateDB session
+    pub ttl: Option<Duration>,
+}
+
+impl PutOptions {
+    const fn default() -> Self {
+        Self { ttl: None }
+    }
+
+    pub(crate) fn expire_ts_from(&self, now: i64) -> Option<i64> {
+        self.ttl.map(|ttl| {
+            now + i64::try_from(ttl.as_millis()).expect(
+                "Duration could not be converted into an i64 timestamp. \
+             Perhaps the duration in millis exceeds the maximum i64?",
+            )
+        })
+    }
+}
+
 /// defines the clock that SlateDB will use during this session
 pub trait Clock {
     /// Returns a timestamp (typically measured in millis since the unix epoch),
     /// must return monotonically increasing numbers (this is enforced
-    /// at runtime and will panic if the invariant is broken)
+    /// at runtime and will panic if the invariant is broken).
     ///
     /// Note that this clock does not need to return a number that
     /// represents the unix timestamp; the only requirement is that
     /// it represents a sequence that can attribute a logical ordering
-    /// to actions on the database
+    /// to actions on the database. Note that in this scenario, it will
+    /// be interpreted as milliseconds in situations where a Duration is
```

Comment:
Ah, I think this might address my other comment. So TTL won't work unless the clock time can convert to millis... 🤔 . Do we have any other use for a clock besides TTL right now? Seems like we might be requiring clock timestamps anyway.,..

Reply:
The alternative would be to have the API specify a `i64` ttl instead of a `Duration` ttl. I was considering this since it would align all of the problems, but having a "duration" feels a lot more natural for expressing TTL.

I think the other use case for timestamps would be some kind of historical/versioned queries (query as of time X and retain copies/don't compact copies until Y time has passed).